### ISSUE TITLE
Shadowing variables inside of flow control don't produce shadowing warning

### ISSUE THREAD

**Issue by Jasper-Bekkers** [microsoft/DirectXShaderCompiler#4029]
This doesn't produce a shadowing warning:

http://shader-playground.timjones.io/f463aa8e1b9249b2dabc84168c522ff7

```hlsl
void stuff(out float3 a) {
     if(true) { 
          float3 a = 1; 
     } 
}
```

Whereas this does:

http://shader-playground.timjones.io/86bc74ba95742f8f45f7f398faa3e9d5

```hlsl
void stuff(out float3 a) { float3 a = 1; }
```

I think the language should either allow variable shadowing all together, or dis-allow it all together, not have exceptions for when variables are shadowed inside if-statements like this.
